### PR TITLE
Adding UI improvements on campaigns table screen

### PR DESCRIPTION
**Pull request by ernestoSk13**
Related to #

## Proposed Changes

*In this PR I'm adding changes to match the new UI proposed by design on the campaign screen

Screenshot 2024-02-05 at 12 44 37

## Testing Instructions

- [ ] Open Calypso with a Woo blog
- [ ] Append the flag `?flags=is_running_in_woo_site`
- [ ] Go to Tools/Advertising/Campaigns
- [ ] Review both web and mobile views
- [ ] Additionally to this, test the segment control hovers. They should be working correctly now and have the same UI as the active control items.

NOTE: I'm using some test properties for conversion rate, 

## Screenshots
Screenshot 2024-02-15 at 12 28 12 p m
(Translators, here are the new sorting fields for campaigns)

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/promote-post-i2/components/campaign-item/index.tsx
Comment (sbarbosa) [inline client/my-sites/promote-post-i2/components/campaign-item/index.tsx]: If you don't have any budget left, I think it's better to show `-` instead of `$0`.

Screenshot 2024-02-06 at 16 21 49

File: client/my-sites/promote-post-i2/components/campaign-item/index.tsx
Comment (sbarbosa) [inline client/my-sites/promote-post-i2/components/campaign-item/index.tsx]: It will be more clear if we rename this function. Right now, it appears to be a variable instead of a function.
Maybe something like `getPostType` will be more descriptive.

File: client/my-sites/promote-post-i2/components/campaigns-table/index.tsx
Comment (sbarbosa) [inline client/my-sites/promote-post-i2/components/campaigns-table/index.tsx]: This needs to be conditional using the WooStore flag. Normal users entering the advertising dashboard won't see the conversion rate.

File: client/my-sites/promote-post-i2/components/campaigns-table/style.scss
Comment (sbarbosa) [inline client/my-sites/promote-post-i2/components/campaigns-table/style.scss:47]: I am unsure if it's in this style line, but right now, we have a big space between clicks and a conversion rate that doesn't feel natural. Checking the designs, I think all of these fields need to be text-align left. 

Screenshot 2024-02-06 at 16 09 31

File: client/my-sites/promote-post-i2/components/campaigns-table/style.scss
Comment (sbarbosa) [inline client/my-sites/promote-post-i2/components/campaigns-table/style.scss]: I don't know if this will change anything, but the conversion rate will be a small number. The max value will be 100% (4 letters). Maybe it is not necessary to have 20% of the width in here when impressions have only 10%.

File: client/my-sites/promote-post-i2/components/search-bar/index.tsx
Comment (sbarbosa) [inline client/my-sites/promote-post-i2/components/search-bar/index.tsx]: It is not necessary to have different options for budget high and low. The order will be set using the `order` prop. 
For example, `SORT_OPTIONS_BUDGET_HIGH_LOW` will be 
```
{
  "orderBy": "budget",
  "order": "desc" 
}
```

You will need to emulate this on the action when the user selects the option in the drawdown (setting both the orderBy and order attribute).

**Edited:**  
I just read the `onCampaignChangeOrderOption` and understood why you do it this way. Butm, it feels a bit off related to the rest of the props, that you use them directly without modifying its value.

I am not sure what could be the best thing to do in here. Maybe we can have the correct value and a separator if we need to specify the order. 
For example: `const SORT_OPTIONS_BUDGET_HIGH_LOW = 'budget|desc';`

This way, we can do a dynamic code in the `onCampaignChangeOrderOption` and check if we have that separator, we already have the order prop.

But, it is not wrong the approach you followed here, you can leave it like this if you prefer.

File: client/my-sites/promote-post-i2/components/search-bar/index.tsx
Comment (sbarbosa) [inline client/my-sites/promote-post-i2/components/search-bar/index.tsx]: We have these options duplicated in the `CampaignsFilter` component. We need to make sure that we don't duplicate this list. It will help us avoid possible bugs in the future.

File: client/my-sites/promote-post-i2/components/search-bar/index.tsx
Comment (sbarbosa) [inline client/my-sites/promote-post-i2/components/search-bar/index.tsx:253]: I checked, and the network call is not being triggered. Maybe we need to also change the hook to accept this new order prop.

File: client/my-sites/promote-post-i2/components/search-bar/index.tsx
Comment (sbarbosa) [inline client/my-sites/promote-post-i2/components/search-bar/index.tsx]: Is it OK to return undefined here? We should always have a default value (at least the backend already is using a default order).

I think it will be better always to show the default value selection, and if it's not one of the options, then add the missing option. Right now the default option is probably date.

**Review by sbarbosa — COMMENTED** [review]
Hello @ernestoSk13 ! I ran a test, reviewed the code, and added a couple of comments.

In addition to those comments, I found the following details:
* We are showing the conversion rate field for all of the cases. We only need to show it to Woo stores. (WPCOM sites won't have a conversion rate yet).

* Just before going to tablet viewport size, the layout breaks a bit
Screenshot 2024-02-06 at 16 11 59
The search box is too small and the table is bigger than the actual screen (probably we have too many columns). Maybe we need to consider going to tablet view a bit sooner here.

* If you don't have any results, the filter dropdown gets partially hidden
Screenshot 2024-02-06 at 16 17 03

**Comment by therocket-gr** [comment]
I was just reviewing this one

@sbarbosa has done a much more thorough review than i did (thanks for that) 

i just want to add that the alignment also feels a bit strange to me 

Screenshot 2024-02-06 at 22 11 55

i am attaching one more screenshot with arrows to make it more obvious

Screenshot 2024-02-06 at 22 11 552

i am not sure if we align all to left that it will fix the issue (maybe center or right should be better - but not sure, we will need to test things out)

also in my case i see the old button (is it supposed to be changed to a arrow??)

Screenshot 2024-02-06 at 22 16 01

File: client/my-sites/promote-post-i2/components/campaign-item/index.tsx
Comment (ernestoSk13) [reply client/my-sites/promote-post-i2/components/campaign-item/index.tsx]: Fixed

File: client/my-sites/promote-post-i2/components/campaigns-table/index.tsx
Comment (ernestoSk13) [reply client/my-sites/promote-post-i2/components/campaigns-table/index.tsx]: Changed

File: client/my-sites/promote-post-i2/components/campaigns-table/style.scss
Comment (ernestoSk13) [reply client/my-sites/promote-post-i2/components/campaigns-table/style.scss:47]: Fixed and improved

File: client/my-sites/promote-post-i2/components/campaigns-table/style.scss
Comment (ernestoSk13) [reply client/my-sites/promote-post-i2/components/campaigns-table/style.scss]: Fixed

File: client/my-sites/promote-post-i2/components/search-bar/index.tsx
Comment (ernestoSk13) [reply client/my-sites/promote-post-i2/components/search-bar/index.tsx]: I have improved this :)

File: client/my-sites/promote-post-i2/components/search-bar/index.tsx
Comment (ernestoSk13) [reply client/my-sites/promote-post-i2/components/search-bar/index.tsx:253]: @sbarbosa  how can we change this?

File: client/my-sites/promote-post-i2/components/search-bar/index.tsx
Comment (ernestoSk13) [reply client/my-sites/promote-post-i2/components/search-bar/index.tsx]: Fixed :)

**Comment by ernestoSk13** [comment]
@sbarbosa  @therocket-gr  I have fixed the problems mentioned above :)

File: client/data/promote-post/types.ts
Comment (jjolmo) [inline client/data/promote-post/types.ts]: I'll follow this rules for the conversion endpoint :D

File: client/my-sites/promote-post-i2/components/search-bar/index.tsx
Comment (jjolmo) [reply client/my-sites/promote-post-i2/components/search-bar/index.tsx:253]: This might help you. This seems triggering correctly in local. I'm unsure if this order prop exists also outside the WooCommerce side, so you might want to add some flag there. 

Add this :
```
if ( searchOptions.order ) {
		searchQueryParams += `&order=${ searchOptions.order.order }&order_by=${ searchOptions.order.orderBy }`;
	}
``` 

in`use-promote-post-campaigns-query-paged.ts` 

I added it like this:
image

This works because we're altering here some content found in useInfiniteQuery:
```
queryKey: [ 'promote-post-campaigns', siteId, searchQueryParams ],
```

This code checks if any of those variables in the array has changed, and if it changes, then it triggers the search again (by calling queryFn). Since we're altering `searchQueryParams` it will trigger the search whenever you change the order selection

**Comment by jjolmo** [comment]
This is how it looks with the conversion data included
image

File: client/my-sites/promote-post-i2/components/campaign-item/index.tsx
Comment (ernestoSk13) [reply client/my-sites/promote-post-i2/components/campaign-item/index.tsx]: Updated

File: client/my-sites/promote-post-i2/components/search-bar/index.tsx
Comment (ernestoSk13) [reply client/my-sites/promote-post-i2/components/search-bar/index.tsx]: Updated

File: client/data/promote-post/types.ts
Comment (ernestoSk13) [reply client/data/promote-post/types.ts]: @sbarbosa  this numbers will be returned as decimals right? and then I need to format them

File: client/my-sites/promote-post-i2/components/search-bar/index.tsx
Comment (ernestoSk13) [reply client/my-sites/promote-post-i2/components/search-bar/index.tsx:253]: Hi :) I have added this logic. We'll need to modify something in DSP to make this work, right?

File: client/data/promote-post/types.ts
Comment (sbarbosa) [reply client/data/promote-post/types.ts]: Yeah, I think we can assume that the format will be the same as the rest of the stats. We have already returned the total clicks and CTR from the DSP.

File: client/my-sites/promote-post-i2/components/search-bar/index.tsx
Comment (sbarbosa) [reply client/my-sites/promote-post-i2/components/search-bar/index.tsx:253]: Hi @ernestoSk13, I checked, and I can't find the logic anywhere in the code. I tested it and do not see the change in the network tab.

I tested locally in my machine with the code Javier shared and it worked Ok. 
Maybe you have some uncommited changes in your local?

**Review by sbarbosa — COMMENTED** [review]
Thanks for the changes, @ernestoSk13 ! 
The code looks good 👍

I checked all the points again, and I am still able to reproduce two of them:
1- The table doesn't fit the screen for desktop screens with a size of `~1057`, and the details button appears to be cut in half. We may need to hide some of the columns to fix the table on those screens correctly.
Screenshot 2024-02-10 at 00 23 46
2- When you change the order by dropdown, we are not fetching the DSP.

File: client/data/promote-post/types.ts
Comment (ozolgu) [reply client/data/promote-post/types.ts]: > @sbarbosa this numbers will be returned as decimals right? and then I need to format them

I think we should return them in a way that clients don't have to format. Not sure about details but just wanted to mention it.

**Comment by jjolmo** [comment]
I had to adapt a bit how we get conversion data. You can see the new changes and some explanation here (I'll copy it since this comes from a priv repo)

 had to change the way we count conversions. It might be possible for a shop to create conversions with different currencies.

So, there could be different conversions_total and conversion_value depending on the currency used. In this PR, I'm adapting the response to return objects instead of numbers

For now, we should show USD in our interfaces. In the future, we will be ready to support them and display them in a list or something similar.

image

This pic shows that we got 4 conversions, but only two of them are in DOLLARS


We will need to change in this PR how conversions are done. I can do it by myself if you prefer @ernestoSk13 . Whatever fits your boat

File: client/my-sites/promote-post-i2/components/search-bar/index.tsx
Comment (ernestoSk13) [reply client/my-sites/promote-post-i2/components/search-bar/index.tsx:253]: hi @sbarbosa  ... yes, that was it... I commited but forgot to push :(

Comment (ernestoSk13) [reply client/my-sites/promote-post-i2/components/search-bar/index.tsx:253]: it is in the remote branch now

**Comment by ernestoSk13** [comment]
> I had to adapt a bit how we get conversion data. You can see the new changes and some explanation here (I'll copy it since this comes from a priv repo)
> 
> had to change the way we count conversions. It might be possible for a shop to create conversions with different currencies.
> 
> So, there could be different conversions_total and conversion_value depending on the currency used. In this PR, I'm adapting the response to return objects instead of numbers
> 
> For now, we should show USD in our interfaces. In the future, we will be ready to support them and display them in a list or something similar.
> 
> image
> 
> This pic shows that we got 4 conversions, but only two of them are in DOLLARS
> 
> We will need to change in this PR how conversions are done. I can do it by myself if you prefer @ernestoSk13 . Whatever fits your boat

hi @jjolmo hi @javiolmo89  i've commited  some changes to adapt to the new conversion values. Let me know if this is a right approach :)

File: client/my-sites/promote-post-i2/components/search-bar/index.tsx
Comment (jjolmo) [inline client/my-sites/promote-post-i2/components/search-bar/index.tsx]: these will be:
`impressions_total`  and `budget_cents` to respect the contract we have with the campaign/search controller

Comment (ernestoSk13) [reply client/my-sites/promote-post-i2/components/search-bar/index.tsx]: Got it, will change it :)

Comment (ernestoSk13) [reply client/my-sites/promote-post-i2/components/search-bar/index.tsx]: changed @jjolmo  :)

Comment (sbarbosa) [reply client/my-sites/promote-post-i2/components/search-bar/index.tsx]: Hi @ernestoSk13 ! I checked and we are sending the order_by/order parameters twice in the URL. 
Maybe they are conflicting with the one from posts, because the date param belongs to posts not campaigns.
And when you load the page, we are fetching the endpoint twice (probably because of the same problem).

Screenshot 2024-02-14 at 12 05 51

I didn't debug it, but maybe we are using the posts default parameters here, or maybe they get mixed before we call the campaigns hook.

Comment (ernestoSk13) [reply client/my-sites/promote-post-i2/components/search-bar/index.tsx]: hi @sbarbosa  I fixed this :)

**Review by sbarbosa — APPROVED** [review]
LGTM 👍

We need to wait for the DSP changes to be deployed before we merge this PR. 
It may be necessary to adapt the names of the `order_by` props. Checking the DSP PR, the `data` field is named differently, but we can apply those changes when the final DSP solution is deployed.

**Comment by sbarbosa** [comment]
Hi @ernestoSk13, I retested the last changes, but I found strange behavior related to the Banner and campaign total badges.

Right now, we are always showing the banner, no matter the amount of campaigns the user has.
Screenshot 2024-02-15 at 15 46 37

**Review by sbarbosa — APPROVED** [review]
Thanks for the change @ernestoSk13!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/11352855

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @ernestoSk13 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11352855).

**Pull request merged**
